catalog/spatial/geocoding/spatial-geocoding-admin-module/src/main/java/org/codice/ddf/spatial/admin/module/service/GeocodingMBean.java

```diff
@@ -0,0 +1,48 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p/>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p/>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.spatial.admin.module.service;
+
+public interface GeocodingMBean {
+
+    /**
+     * Calls {@link org.codice.ddf.spatial.geocoding.GeoEntryIndexer#updateIndex}
+     * with the given URL and country code.
+     * @param url - the URL of the GeoNames server
+     * @param ccode - the country code of the file to download
+     * @param createIndex - true if the index is to be created, false if the index is to be appended to
+     *
+     * @return true on successful execution, false when an error occurs in the GeoEntryIndexer
+     */
+    boolean execWithUrl(String url, String ccode, String createIndex);
```

Comment:
Seems like we could come up with some better names for these methods. I really have no idea what these are doing by just reading them.